### PR TITLE
Backup: Add render switch to validate multisites and backup feature on backup page

### PR DESCRIPTION
**Pull request by Initsogar**
If a customer is trying to access the backup page directly in Calypso using a multisite with backup capabilities, they are getting a message saying "WordPress multi-sites are not supported".

This is because when accessing directly the page, we are not querying site features before. So the idea of this PR is to add a new switch component to validate if a site is part of a Jetpack multisite that can be used as part of the [showUnavailableForMultisites()](https://github.com/Automattic/wp-calypso/blob/0650cc21ec42c79e8c1536a4966d60976d0cf5a4/client/my-sites/backup/controller.js#L97) middleware.
## Proposed Changes

* Add new switch component to validate if a site is part of a Jetpack multisite
* Use new switch component on `showUnavailableForMultisites` backup controller middleware.

## Testing Instructions

First, we should grab a site that is part of a multisite and also has (or not) the backup feature. You can test this PR in your local environment by returning `true` on [`isJetpackSiteMultiSite` selector](https://github.com/Automattic/wp-calypso/blob/11e5d35381f271cf1c88d7af1f8d45cb820c1dc5/client/state/sites/selectors/is-jetpack-site-multi-site.js#L12) to simulate the multisite condition. 

Then navigate directly to http://jetpack.cloud.localhost:3000/backup/{site_slug}.

If your site has no backup feature, you should see the following message:

| Jetpack Cloud | Calypso Blue |
|---|---|
| image | image |

If your site has the backup feature, you should see the backup section as usual:

image

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/backup/controller.js [hunk 1]
```diff
@@ -97,21 +96,21 @@ export function showUnavailableForVaultPressSites( context, next ) {
...
-			<BackupUpsell reason="multisite_not_supported" />
-		) : (
-			<WPCOMBackupUpsell reason="multisite_not_supported" />
-		);
-	}
+	context.primary = (
+		<IsMultiSiteSwitch
+			trueComponent={ message }
+			falseComponent={ context.primary }
+			loadingComponent={ <p>Loading..</p> }
```
Comment (Initsogar) [inline client/my-sites/backup/controller.js]: Let's use a loading placeholder instead.

File: client/my-sites/backup/is-multisite-switch.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import { isJetpackSiteMultiSite } from 'calypso/state/sites/selectors';
+import getSelectedSiteId from 'calypso/state/ui/selectors/get-selected-site-id';
+
+type Props = {
+	trueComponent: ReactNode;
+	falseComponent: ReactNode;
+	loadingComponent?: ReactNode;
+};
+
+const IsMultiSiteSwitch: FunctionComponent< Props > = ( {
```
Comment (jwebbdev) [inline client/my-sites/backup/is-multisite-switch.tsx]: It's a bit wordy, but I wonder if something like `IsMultiSiteNoPlanSwitch` or similar might be a less misleading name.

**Review by mavegaf — APPROVED** [review]
Tested and works as expected 👍

File: client/my-sites/backup/multisite-no-backup-plan-switch.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+const MultisiteNoBackupPlanSwitch: FunctionComponent< Props > = ( {
+	trueComponent,
+	falseComponent,
+	loadingComponent,
+} ) => {
+	const siteId = useSelector( getSelectedSiteId );
+
+	const isMultiSite =
+		useSelector( ( state ) => siteId && isJetpackSiteMultiSite( state, siteId ) ) || false;
```
Comment (macbre) [inline client/my-sites/backup/multisite-no-backup-plan-switch.tsx:26]:
```suggestion
		useSelector( ( state ) => siteId && isJetpackSiteMultiSite( state, siteId ) );
```

A left-over from debugging?

**Comment by macbre** [comment]
@Initsogar - thanks for your help here!

File: client/my-sites/backup/multisite-no-backup-plan-switch.tsx (hunk 3 above)
Comment (Initsogar) [reply client/my-sites/backup/multisite-no-backup-plan-switch.tsx:26]: @macbre The `isJetpackSiteMultiSite` selector can also be `null`, but for this use case it would be better to always return a boolean, so in case it is null I'm returning false, so we could use it for the `renderCondition` prop in `RenderSwitch`. Maybe there's a better way to handle it in TS? Thoughts?

Comment (macbre) [reply client/my-sites/backup/multisite-no-backup-plan-switch.tsx:26]: Oh, then this code does make sense.
